kivymd/uix/button.py

```diff
@@ -2558,46 +2650,96 @@ def __init__(self, **kwargs):
+            self._has_text = True
+        if self._is_filled is None:
+            self._is_filled = False
 
 
 # ------------------------------------------------------------------------------
-# BaseFlatIconButton
+# MDRectangleFlatButton
 # ------------------------------------------------------------------------------
-#
-# Builder.load_string(
-#     """
-#
-#     """,
-#     filename="MDBaseFlatIconButton.kv",
-# )
-# class BaseFlatIconButton(MDFlatButton):
-# icon = StringProperty("android")
-# """
-# Button icon.
-#
-# :attr:`icon` is an :class:`~kivy.properties.StringProperty`
-# and defaults to `'android'`.
-# """
 
-# text = StringProperty("")
-# """Button text.
-#
-# :attr:`text` is an :class:`~kivy.properties.StringProperty`
-# and defaults to `''`.
-# """
 
-# show_label = BooleanProperty(False)
+class MDRectangleFlatButton(BaseOutlineButton, MDFlatButton):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __after_init__(self, *args):
+        if self.corner_type is None:
+            self.corner_type = "Square"
+        super().__after_init__(*args)
+
+
+# ------------------------------------------------------------------------------
+# MDRectangleFlatIconButton
+# ------------------------------------------------------------------------------
+
+
+class MDRectangleFlatIconButton(MDRectangleFlatButton, icon_behavior):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __after_init__(self, *args):
+        super().__after_init__(*args)
+        # self.theme_button_color = "Primary" if self.theme_button_color is None else "Custom"
+
+
+# ------------------------------------------------------------------------------
+# MDRoundFlatButton
+# ------------------------------------------------------------------------------
+
+
+class MDRoundFlatButton(MDRectangleFlatButton):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __after_init__(self, *dt):
+        if self.corner_type is None:
+            self.corner_type = "Rounded"
+        super().__after_init__(*dt)
+
+
+# ------------------------------------------------------------------------------
+# MDRoundFlatIconButton
+# ------------------------------------------------------------------------------
+
+
+class MDRoundFlatIconButton(MDRoundFlatButton, icon_behavior):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __after_init__(self, *dt):
+        super().__after_init__(*dt)
+
+
+# ------------------------------------------------------------------------------
+# MDBevelFlatButton
+# ------------------------------------------------------------------------------
+
+
+class MDBevelFlatButton(MDRectangleFlatButton):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __after_init__(self, *dt):
+        if self.corner_type is None:
+            self.corner_type = "Bevel"
+        super().__after_init__(*dt)
+
+
```

Comment:
the classes have been relocated, now they, the order is base class, corner type and then icon version